src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/PurposePersistentBehavior.scala

```diff
@@ -376,6 +375,28 @@ object PurposePersistentBehavior {
     riskAnalysisUpdated = riskAnalysisOpt.map(PersistentPurposeVersionDocument.fromAPI).orElse(version.riskAnalysis)
     versionWithRisk     = version.copy(riskAnalysis = riskAnalysisUpdated)
     _ <- versionValidation(versionWithRisk)
-  } yield updatePurposeFromState(purpose, versionWithRisk, newState, stateChangeDetails)(dateTimeSupplier)
+  } yield updatePurposeFromState(
+    purpose,
+    versionWithRisk,
+    getPurposeState(purpose, version.state, newState),
+    stateChangeDetails
+  )(dateTimeSupplier)
 
+  private val notFinalStates: Set[PersistentPurposeVersionState] = Set[PersistentPurposeVersionState](Active, Suspended)
+
+  private def getPurposeState(
+    purpose: PersistentPurpose,
+    currentState: PersistentPurposeVersionState,
+    newState: PersistentPurposeVersionState
+  ): PersistentPurposeVersionState = {
+
+    val actualValidState: Option[PersistentPurposeVersionState] =
+      purpose.versions.map(_.state).find(notFinalStates.contains)
+
+    currentState match {
+      case WaitingForApproval if actualValidState.nonEmpty => actualValidState.getOrElse(newState)
```

Comment:
I don't remember if a `WaitingForApproval` version can remain in a `WaitingForApproval` state (maybe after changing the expected load? is it possible? I'm not sure).

In case, could it be safer changing this check into this?
```suggestion
      case WaitingForApproval if newState != WaitingForApproval && actualValidState.nonEmpty => actualValidState.getOrElse(newState)
```

(btw to me this pattern matching seems more like an `if`)

Reply:
Ok for the IF.
I add also the `newState != WaitingForApproval` check to enforce the constraint but it is not necessary, in fact in the case of updating a waiting-for-approval version, there is a specific command that uses a different flow.